native/Avalonia.Native/src/OSX/main.mm

```diff
@@ -133,6 +138,26 @@ virtual HRESULT SetShowInDock(int show)  override
         }
     }
     
+    virtual HRESULT SetPreventAppNap(bool preventAppNap) override
+    {
+        START_COM_CALL;
+        
+        @autoreleasepool
+        {
+            if (preventAppNap && s_preventAppNapHandle == nullptr)
+            {
+                NSActivityOptions options = NSActivityUserInitiatedAllowingIdleSystemSleep;
+                s_preventAppNapHandle = [[NSProcessInfo processInfo] beginActivityWithOptions:options reason:@"User requested to prevent App Nap."];
```

Comment:
We need to keep a reference to the result of `beginActivityWithOptions`. App Nap will turn on again when this is deallocated.

For this reasom `s_preventAppNapHandle` is in an anomymous namespace at the tom (equivalent to static).

Question: Do we need thread safety here? Or should this only be called from main thread anyway? What is the policy?